tests/Api/PersistentSpec.hs

```diff
@@ -125,6 +125,25 @@ spec = do
     let getIdeasWithTopic = filter (isJust . view ideaTopic) <$> getIdeas
     findAllInBySpec "findIdeasByTopicId" getIdeasWithTopic findIdeasByTopicId (ideaTopic . _Just) changeAUID
 
+    describe "addIdeaSpace" $ do
+        let test :: (Int -> Int) -> IdeaSpace -> SpecWith (Persist :~> IO)
+            test upd ispace = do
+                it ("can add " <> showIdeaSpace ispace) $ \(Nat rp) -> do
```

Comment:
This is like BDD: Given something, do something, test someting based on the previous and after state. No if in testcase. Just high level combinators! Big like! :)